test/e2e/registry.go

```diff
@@ -0,0 +1,98 @@
+package e2e
+
+import (
+	"fmt"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"os/exec"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// This module contains helper functions for copying images and creating image registries
+// Use for tests that require manipulating images or use of custom container registries
+
+const (
+	registryImage = "registry:2.7.1"
+	registryName  = "registry"
+	localFQDN     = "localhost:5000"
+)
+
+func createDockerRegistry(client operatorclient.ClientInterface, namespace string) (string, error) {
+	registry := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      registryName,
+			Namespace: namespace,
+			Labels:    map[string]string{"name": registryName},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  registryName,
+					Image: registryImage,
+					Ports: []corev1.ContainerPort{
+						{
+							HostPort:      int32(5000),
+							ContainerPort: int32(5000),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      registryName,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"name": registryName},
+			Ports: []corev1.ServicePort{
+				{
+					Port: int32(5000),
+				},
+			},
+		},
+	}
+
+	_, err := client.KubernetesInterface().CoreV1().Pods(namespace).Create(registry)
+	if err != nil {
+		return "", fmt.Errorf("creating test registry: %s", err)
+	}
+
+	_, err = client.KubernetesInterface().CoreV1().Services(namespace).Create(svc)
+	if err != nil {
+		return "", fmt.Errorf("creating test registry: %s", err)
+	}
+
+	// <name>.<namespace>.svc.cluster.local
+	// return strings.Join([]string{registryName, namespace, "svc.cluster.local:5000"}, "."), nil
+
+	// open port 5000 to the registry for skopeo copy to work
+	// ensure pod is ready before attempting port-forwarding
+	time.Sleep(5 * time.Second)
```

Comment:
It might be worthwhile to replace the sleep with some kind of retry strategy, to cut down on flakes caused by registry pod slowness.

Reply:
I agree, but I haven't seen this test flake out on CI once yet. Maybe we can revisit if we do see a flake occur?

Reply:
Ok I can add an `awaitPod` here, just so the local version is a bit cleaner.